node/runtime/src/lib.rs

```diff
@@ -86,6 +87,28 @@ impl Convert<u128, u128> for CurrencyToVoteHandler {
 	fn convert(x: u128) -> u128 { x * Self::factor() }
 }
 
+pub struct WeightToFeeHandler;
+
+impl Convert<Weight, Balance> for WeightToFeeHandler {
+	fn convert(weight: Weight) -> Balance {
+		// 40/1000000 = 4/100000 = 4/10^5 = 0.00004
+		let variability_fee = Permill::from_parts(40);
+		// 0.00004^2 = 16/10^10 ~= 2/10^9
+		let variability_fee_squared = Perbill::from_parts(2);
+		let potential_weight = <system::Module<Runtime>>::all_extrinsics_weight() + weight;
+		let high_fee: bool =  IDEAL_TRANSACTIONS_WEIGHT <= potential_weight;
+		// workaround because unsigned cannot be negative
+		let diff = if high_fee { potential_weight - IDEAL_TRANSACTIONS_WEIGHT }
```

Comment:
I think you should also think about the fact that you can technically make the `weight` simply signed. Not sure if it will make things easier for you or not.